cogs/birthday.py

```diff
@@ -0,0 +1,126 @@
+        
+        return all_records
+
+    @tasks.loop(minutes=59)
+    async def check_birthday(self):
+        '''
+        Checks json for birthday, if match, send message with discord names
+        '''
+        now = datetime.datetime.now()
+        current_date = now.strftime("%m-%d")
+        date_formatted = now.strftime('%B %d') + ('st' if now.day in [1, 21, 31] else 'nd' if now.day in [2, 22] else 'rd' if now.day in [3, 23] else 'th')
+        
+        if not now.hour == 8:
+            return
+
+        records = self.get_airtable_data()
+        if records is not None:
+            channel = self.bot.get_channel(self.MEMBERSHIP_CHANNEL_ID)
+            response_title = f"<a:blobcake:1063101478321000558> Today's Birthdays: {date_formatted}"
+            response_description = "List of INIT Members that are celebrating their birthday's today:"
+            embed_color = YELLOW_COLOR
+            embed_response = discord.Embed(title=response_title, description=response_description, color=embed_color)
+            birthday_count = 0
+
+            for record in records:
+                if 'Birthday' not in record['fields'] or 'Discord User Name' not in record['fields']:
+                    continue
+                    
+                birthday = record['fields']['Birthday']
+                if birthday[5:10] == current_date:
+                    discord_user_name = record['fields']['Discord User Name']
+                    discord_user = discord.utils.get(self.bot.guilds, name=discord_user_name)
+                    first_name = record['fields']['First Name']
+                    last_name = record['fields']['Last Name']
+                    embed_response.add_field(name=f'{first_name} {last_name}', value=f'({discord_user.mention if discord_user is not None else discord_user_name})', inline=True)  # '\u200b' is a zero-width space to set an empty value
+                    birthday_count += 1
+                    
+            embed_response.set_footer(text=f'\n\nWrite them something special in the Spotlight channel')
+            
+        else:
+            channel = self.bot.get_channel(self.BOT_CHANNEL_ID)
+            response_title = f"<a:utilfailure:809713365088993291> Failed to load birthdays"
+            response_description = "Failed to load membership records"
+            embed_color = RED_COLOR
+            embed_response = discord.Embed(title=response_title, description=response_description, color=embed_color)
+        
+        if birthday_count == 0:
+            response_title = f"<a:blobcake:1063101478321000558> Today's Birthdays: {date_formatted}"
+            response_description = "None of our members are celebrating their birthdays today."
+            embed_color = YELLOW_COLOR
+            embed_response = discord.Embed(title=response_title, description=response_description, color=embed_color)
+            await channel.send(embed=embed_response)
```

Comment:
This feels like it'll end up being spam since I'm pretty sure most days we won't be celebrating birthdays. Maybe it might be good to just message the bot channel every day that the birthday task ran and how many birthdays were found. At least there we can use it as a canary to see if the birthdays check was missing for a day.

Reply:
Yeah, the message for no bdays can be sent in bot log. @ginorey I think we could also benefit from pinging membership cabinet role with this message. We can always revert this but I can see the message being ignored with the current engagement we are having.

@mrmcpowned Interestingly, we actually have like 5+ birthdays every day! (The table has more than 4000 entries so it kinda makes sense haha).

Reply:
went ahead and changed this to send it to the bot channel instead \